pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -168,10 +168,7 @@ def advance(self, *args: Any, **kwargs: Any) -> None:
             batch_output = []
         else:
             # hook
-            response = self.trainer.call_hook("on_batch_start")
-            if response == -1:
-                self.batch_progress.increment_processed()
-                raise StopIteration
+            self.trainer.call_hook("on_batch_start")
```

Comment:
I think we should just deprecate and remove this hook. doesn't provide any additional significance. We already have `on_train_batch_start`.

Reply:
You should open an RFC about it in that case